### PR TITLE
codec: improve the coverage of key path in codec

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/15634

Problem Summary:

### What is changed and how it works?

What's Changed:

add test for cover more type

How it Works:

### Related changes

- n/a

### Check List 

Tests 

- Unit test

Side effects

- n/a

### Release note 

- n/a

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/17023)

### CODE REVIEW

File: util/codec/codec_test.go [hunk 1]
```diff
@@ -1198,7 +1251,7 @@ func (s *testCodecSuite) TestHashChunkColumns(c *C) {
 	}
 
 	// Test hash value of every single column that is not `Null`
-	for i := 2; i < len(tps); i++ {
+	for i := 13; i < len(tps); i++ {
```
Comment (wshwsh12) [inline util/codec/codec_test.go:1254]:
```suggestion
	for i := 12; i < len(tps); i++ {
```

File: util/codec/codec_test.go [hunk 2]
```diff
@@ -1176,17 +1227,19 @@ func (s *testCodecSuite) TestHashChunkColumns(c *C) {
...
-	// Test hash value of the first two `Null` columns
-	for i := 0; i < 2; i++ {
+	// Test hash value of the first 12 `Null` columns
+	for i := 0; i < 12; i++ {
 		c.Assert(chk.GetRow(0).IsNull(i), Equals, true)
 		err1 := HashChunkColumns(sc, vecHash, chk, tps[i], i, buf, hasNull)
-		err2 := HashChunkRow(sc, rowHash[0], chk.GetRow(0), tps, colIdx[i:i+1], buf)
-		err3 := HashChunkRow(sc, rowHash[1], chk.GetRow(1), tps, colIdx[i:i+1], buf)
-		err4 := HashChunkRow(sc, rowHash[2], chk.GetRow(2), tps, colIdx[i:i+1], buf)
+		err2 := HashChunkSelected(sc, vecHash, chk, tps[i], i, buf, hasNull, make([]bool, len(datums)))
```
Comment (wshwsh12) [inline util/codec/codec_test.go:1234]:
```suggestion
		err2 := HashChunkSelected(sc, vecHash, chk, tps[i], i, buf, hasNull, nil)
```
So all the values are selected ?

Comment (lysu) [reply util/codec/codec_test.go]: emmm.... I just want to cover `sel != nil && sel[i] = false` condition, do some modification, maybe better now, ptal if free thx~

**Pull request merged**